bagri-server/bagri-server-hazelcast/src/main/java/com/bagri/server/hazelcast/impl/DocumentManagementImpl.java

```diff
@@ -1303,8 +1304,36 @@ public int removeDocumentFromCollections(String uri, String[] collections) {
 		return remCount;
 	}
 
-	public void deleteDocumentFromContentCache(Map.Entry<DocumentKey, Document> entry) {
-		cntCache.delete(entry.getKey());
+	public Object processDocumentRemoval(Map.Entry<DocumentKey, Document> entry, Properties properties) {
+		Document doc = entry.getValue();
+		Transaction tx = txManager.getCurrentTransaction();

```

Comment:
no, we can't get transaction here, as we'll get exception from HZ layer. all the logic about exception throwing better to leave in the processor itself and do all the rest (from getDocument(lastKey) call) here